recipes/autoconf/all/conanfile.py

```diff
@@ -9,30 +11,37 @@ class AutoconfConan(ConanFile):
     description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
     topics = ("conan", "autoconf", "configure", "build")
     license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
-    exports_sources = "patches/**"
     settings = "os", "arch", "compiler"
+
+    exports_sources = "patches/**"
+
     _autotools = None
 
     @property
     def _source_subfolder(self):
         return os.path.join(self.source_folder, "source_subfolder")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
     def requirements(self):
         self.requires("m4/1.4.18")
 
+    def build_requirements(self):
+        self.build_requires("m4/1.4.18")
```

Comment:
@SpaceIm 
You have gathered a lot of cross building experience in the past months.
The bug I described in #6673 and does not fix in this pr -> do you think it is a conan or recipe bug?

Reply:
I'm not sure to fully understand the issue. We are talking of cross-building autoconf with 2 profiles, right? Not having autoconf in build requirements in 2 profiles context.

You should check 2 profiles:

```python
if hasattr(self, "settings_build"):
    self.build_requires("m4/1.4.18")
```

Reply:
Yes, cross-building `autoconf` with 2 profiles that happen to be the same.
Adding that test always succeeds in 2 profile approach.
It's like, when using the 2 profile approach, the `m4` build requirement is not applied.